tests/src/PatternAssertion/PaginationPatternAssert.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_bootstrap_theme\PatternAssertion;
+
+use Symfony\Component\DomCrawler\Crawler;
+
+/**
+ * Assertions for the pagination pattern.
+ */
+class PaginationPatternAssert extends BasePatternAssert {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getAssertions(string $variant): array {
+    return [
+      'links' => [
+        [$this, 'assertLinks'],
+      ],
+      'alignment' => [
+        [$this, 'assertAlignment'],
+      ],
+      'size' => [
+        [$this, 'assertSize'],
+      ],
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function assertBaseElements(string $html, string $variant): void {
+    $this->assertCounts([
+      'nav' => 1,
+      'ul' => 1,
+    ], new Crawler($html));
+  }
+
+  /**
+   * Asserts the pager links.
+   *
+   * @param array[] $expected
+   *   Expected pager links.
+   *   Each link is an array, with the following keys:
+   *     - 'url': The link destination.
+   *     - 'label': The label, if exists.
+   *     - 'icon': The name of the icon.
+   *     - 'active': Whether the item has an 'active' class.
+   * @param \Symfony\Component\DomCrawler\Crawler $crawler
+   *   The crawler.
+   */
+  protected function assertLinks(array $expected, Crawler $crawler): void {
+    $expected_bcl_icon_url = '/' . \Drupal::service('extension.list.theme')->getPath('oe_bootstrap_theme') . '/assets/icons/bcl-default-icons.svg';
+
+    $actual_links_data = array_map(
+      function (\DOMElement $actual_li_element) use ($expected_bcl_icon_url) {
```

Comment:
Yep.

Btw, PhpStorm asks me to add `ext-dom` to composer.json.
I won't, because it is already required by other packages, and I think we don't do this elsewhere.